test/functional/mining_basic.py

```diff
@@ -245,6 +247,15 @@ def chain_tip(b_hash, *, status='headers-only', branchlen=1):
         node.submitheader(hexdata=CBlockHeader(bad_block_root).serialize().hex())
         assert_equal(node.submitblock(hexdata=block.serialize().hex()), 'duplicate')  # valid
 
+        # Generate a block with witness data and round-trip it using python framework
+        block_hash = node.generatetoaddress(1, node.getnewaddress())[0]
```

Comment:
You're apparently not supposed to use `getnewaddress` (a wallet call) in `mining_basic.py` because it is run even in builds where wallet is disabled.

Reply:
hmmm, maybe we should disable wallet for the nodes?

Reply:
ah I see `node.get_deterministic_priv_key().address`

Reply:
fixed.